core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -431,54 +436,41 @@ public void close(ScheduledExecutorService instance) {
    */
   @VisibleForTesting
   static class TimeoutMarshaller implements Metadata.AsciiMarshaller<Long> {
+    private static final BiMap<TimeUnit, Character> UNITS =
```

Comment:
It isn't well-known that you are depending on the iteration order. Also, you are depending on the immutable implementation, not the `BiMap` interface. If I changed the construction to `HashBiMap` I would be surprised that it would break, as typically it would be considered mostly interchangeable.